internal/cmd/deploy/deploy.go

```diff
@@ -0,0 +1,187 @@
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.name, "name", "", "Service name")
+
+	return cmd
+}
+
+func runDeploy(f *cmdutil.Factory, opts *Options) error {
+	if f.Interactive {
+		return runDeployInteractive(f, opts)
+	} else {
+		return runDeployNonInteractive(f, opts)
+	}
+}
+
+func runDeployNonInteractive(f *cmdutil.Factory, opts *Options) error {
+	repoOwner, repoName, err := getRepoInfo()
+	if err != nil {
+		return err
+	}
+
+	repoID, err := getRepoID(repoOwner, repoName)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println(repoID)
+
+	return nil
+}
+
+func runDeployInteractive(f *cmdutil.Factory, opts *Options) error {
+	s := spinner.New(cmdutil.SpinnerCharSet, cmdutil.SpinnerInterval,
+		spinner.WithColor(cmdutil.SpinnerColor),
+		spinner.WithSuffix(" Fetching repository informations..."),
+	)
+	s.Start()
+	repoOwner, repoName, err := getRepoInfo()
+	if err != nil {
+		return err
+	}
+
+	// Use repo name as default service name
+	if opts.name == "" {
+		opts.name = repoName
+	}
+
+	// Get repo ID and branches concurrently
+	repoIDChan := make(chan int)
+	branchesChan := make(chan []string)
+	errChan := make(chan error)
+
+	go func() {
+		repoID, err := getRepoID(repoOwner, repoName)
+		if err != nil {
+			errChan <- err
+			return
+		}
+		repoIDChan <- repoID
+	}()
+
+	go func() {
+		branches, err := getRepoBranches(repoOwner, repoName)
+		if err != nil {
+			errChan <- err
+			return
+		}
+		branchesChan <- branches
+	}()
+
+	repoID := <-repoIDChan
+	branches := <-branchesChan
+
+	// Check for errors
+	select {
+	case err := <-errChan:
+		return err
+	default:
+	}
+
+	s.Stop()
+
+	// If repo has only one branch, use it as default branch
+	// Otherwise, ask user to select a branch
+	var branch string
+
+	if len(branches) == 1 {
+		branch = branches[0]
+	} else {
+		_, err = f.Prompter.Select("Select branch", branch, branches)
+		if err != nil {
+			return err
+		}
+	}
+
+	s = spinner.New(cmdutil.SpinnerCharSet, cmdutil.SpinnerInterval,
+		spinner.WithColor(cmdutil.SpinnerColor),
+		spinner.WithSuffix(" Creating service..."),
+		spinner.WithFinalMSG(cmdutil.SuccessIcon+" Service created 🥂\n"),
+	)
+	s.Start()
+
+	_, err = f.ApiClient.CreateService(context.Background(), f.Config.GetContext().GetProject().GetID(), opts.name, repoID, branch)
+	if err != nil {
+		return err
+	}
+	s.Stop()
+
+	return nil
+}
+
+func getRepoInfo() (string, string, error) {
+	cmd := exec.Command("git", "remote", "get-url", "origin")
+	cmd.Dir = "."
+	out, err := cmd.Output()
+	if err != nil {
+		return "", "", err
+	}
+
+	repoURL := strings.TrimSpace(string(out))
+	parts := strings.Split(repoURL, "/")
+	if len(parts) < 2 {
+		return "", "", fmt.Errorf("invalid repository URL")
+	}
+
+	repoOwner := strings.TrimPrefix(parts[len(parts)-2], "git@github.com:")
+	repoName := strings.TrimSuffix(parts[len(parts)-1], ".git")
+
+	return repoOwner, repoName, nil
+}
+
+func getRepoID(repoOwner string, repoName string) (int, error) {
+	client := github.NewClient(nil)
```

Comment:
You can left a TODO here: "deal with GitHub Auth, reading `GITHUB_TOKEN` env and set HTTP client header."